packages/slate-sections-plugin/index.js

```diff
@@ -0,0 +1,204 @@
+      const combinedLocales = files.includes('locales')
+        ? await this._combineLocales(path.resolve(directoryPath, 'locales'))
+        : null;
+
+      const schemaContent = combinedLocales
+        ? await this._createSchemaContentWithLocales(
+            combinedLocales,
+            path.resolve(directoryPath, 'schema.json'),
+          )
+        : await fs.readFile(path.resolve(directoryPath, 'schema.json'));
+      const schemaSource = new RawSource(
+        `{% schema %}\n${schemaContent}{% endschema %}`,
+      );
+      this._addLiquidFileToAssets(liquidSourcePath, compilation, schemaSource);
+    } else {
+      this._addLiquidFileToAssets(liquidSourcePath, compilation);
+    }
+  }
+
+  /**
+   * Gets all the translations for a translation key
+   *
+   * @param {*} key The key of the value to receive within the locales json object
+   * @param {*} localizedSchema Object containing all the translations in locales
+   * @returns Object with index for every language in the locales folder
+   */
+  _getLocalizedValues(key, localizedSchema) {
+    const combinedTranslationsObject = {};
+    Object.keys(localizedSchema).forEach((language) => {
+      combinedTranslationsObject[language] = _.get(
+        localizedSchema[language],
+        key,
+      );
+    });
+    return combinedTranslationsObject;
+  }
+
+  /**
+   * Goes through the main schema to get the translation keys and to fill the schema with translations
+   *
+   * @param {*} localizedSchema The schema with the combined locales
+   * @param {*} mainSchemaPath The path to the main schema (schema.json)
+   * @returns
+   */
+  async _createSchemaContentWithLocales(localizedSchema, mainSchemaPath) {
+    const traverse = (obj) => {
+      for (const i in obj) {
+        if (typeof obj[i].t === 'string') {
+          obj[i] = this._getLocalizedValues(obj[i].t, obj, localizedSchema);
+        } else if (typeof obj[i] === 'object') {
+          traverse(obj[i]);
+        }
+      }
+      return JSON.stringify(obj);
+    };
+    const mainSchema = JSON.parse(await fs.readFile(mainSchemaPath, 'utf-8'));
+    return traverse(mainSchema);
+  }
+
+  /**
+   * Creates a single JSON object from all the languages in locales
+   *
+   * @param {*} localesPath Absolute path to the locales folder /sections/section-name/locales/
+   * @returns
+   */
+  async _combineLocales(localesPath) {
+    const localesFiles = await fs.readdir(localesPath);
+    const jsonFiles = localesFiles.filter((fileName) =>
+      fileName.endsWith('.json'),
+    );
+
+    const accumulator = {};
+    await Promise.all(
+      jsonFiles.map(async (file) => {
+        const localeCode = path
+          .basename(file)
+          .split('.')
+          .shift();
+        const fileContents = JSON.parse(
+          await fs.readFile(path.resolve(localesPath, file), 'utf-8'),
+        );
+        accumulator[localeCode] = fileContents;
+      }),
+    );
+
+    return accumulator;
```

Comment:
I think you could alternatively write:
```
return await jsonFiles.reduce(async (promise, file) => {
  const accumulator = await promise;
  ...
  return accumulator[localCode] = await fs.readJson(path.resolve(localesPath, file));
}, Promise.resolve({}));
```

Let me know if it works? 😅

Reply:
Worked 👍